src/sync.cpp

```diff
@@ -57,7 +57,7 @@ struct CLockLocation {
     {
         return strprintf(
             "%s %s:%s%s (in thread %s)",
```

Comment:
Now that the third parameter is of type `int` and not `std::string` anymore, the format string also has to be adapted?
```suggestion
            "%s %s:%d%s (in thread %s)",
```

Reply:
tinyformat.h is a type safe printf replacement library in a single C++ header file. If you've ever wanted printf("%s", s) to just work regardless of the type of s, tinyformat might be for you.

https://github.com/c42f/tinyformat#a-minimal-type-safe-printf-replacement

Reply:
Ah, that's nice and convenient. 
Is it smart though that the function has exactly the same name as the one from the standard library then (with not even a namespace in front)? As reader I'd hardly come to the idea that this would behave different than the standard implementation and be more forgiving.

Reply:
I don't like the name either (I prefer `tfm::format`), but I think the name differs from the std lib one:
```
strprintf
sprintf

Reply:
Oops -- you are right! I missed the `tr` and read it indeed as `sprintf`. :walking_man: